src/startupscriptgenerator/node/scriptgenerator.go

```diff
@@ -52,22 +53,28 @@ func (gen *NodeStartupScriptGenerator) GenerateEntrypointScript() string {
 	// Expose the port so that a custom command can use it if needed.
 	common.SetEnvironmentVariableInScript(&scriptBuilder, "PORT", gen.BindPort, DefaultBindPort)
 
-	// If a file called node_modules.zip is found, we consider it to be
-	// the zipped contents of the app's node_modules folder. We unzip it
-	// at the root level, so node runtime can still find it, and
-	// it is not persisted in a shared network volume where the app is.
-	const nodeModules string = "node_modules"
-	const nodeModulesFile string = nodeModules + ".tar.gz"
-	scriptBuilder.WriteString("if [ -f " + nodeModulesFile + " ] && [ ! \"$ORYX_DISABLE_NODE_MODULES_EXTRACTION\" == \"true\" ]; then\n")
-	scriptBuilder.WriteString("    echo \"Found '" + nodeModulesFile + "', will extract its contents as node modules.\"\n")
-	scriptBuilder.WriteString("    echo \"Removing existing modules directory...\"\n")
-	scriptBuilder.WriteString("    rm -fr /" + nodeModules + "\n")
-	scriptBuilder.WriteString("    mkdir -p /" + nodeModules + "\n")
-	scriptBuilder.WriteString("    echo \"Extracting modules...\"\n")
-	scriptBuilder.WriteString("    tar -xzf " + nodeModulesFile + " -C /" + nodeModules + "\n")
-	scriptBuilder.WriteString("    echo \"Done.\"\n")
-	scriptBuilder.WriteString("fi\n\n")
-
+	if !gen.SkipNodeModulesExtraction {
+		const oryxManifestFile string = "oryx-manifest.toml"
```

Comment:
since this path appears in C# too, it could be consolidated into the `file-paths` section of the `build-constants.yaml`.